catalogue_pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/modules/ExecutionContextModule.scala

```diff
@@ -0,0 +1,15 @@
+package uk.ac.wellcome.platform.idminter.modules
+
+import com.google.inject.Provides
+import com.twitter.inject.{Injector, TwitterModule}
+import javax.inject.Singleton
+import uk.ac.wellcome.platform.idminter.GlobalExecutionContext
+
+import scala.concurrent.ExecutionContext
+
+object ExecutionContextModule extends TwitterModule {
+  @Provides
+  @Singleton
+  def provideExecutionContext(injector: Injector): ExecutionContext =
```

Comment:
Since we are writing Guice providers, could we make it get the excution context from the `ActorSystem` rather than from the `GlobalExecutionContext`? Somethink like 
```
def provideExecutionContext(actorSystem: ActorSystem): ExecutionContext = actorSystem.dispatcher
```
I think it would be good to deprecate those `GlobalExecutionContext` and this might be a step towards that

Reply:
I generally agree, but making that change in this patch might be a step too far. I'd prefer to make this update in a future patch.

Reply:
definitely don't get rid of the GlobalExecutionContexts here, but I would avoid adding an extra usage of them in this patch. Making the provider use the actor system will at least not worsen the position we are in now. Also, delegating all scheduling responsibilities to the actor system is probably more reliable